### ISSUE TITLE
Firefox Issue

### ISSUE THREAD

**Issue by hassandraga** [alifcommunity/webui#85]
If you use Linux, and your only installed web browser is the latest Firefox, WebUI probably won't work because Firefox itself can't create a new profile.

`firefox -CreateProfile "WebUI /home/me/.WebUI/WebUIFirefoxProfile"`
```
Gtk-Message: 20:15:00.605: Not loading module "atk-bridge":
The functionality is provided by GTK natively. Please try to not load it.
Error creating profile.
```

**Comment by malisipi** [comment]
![image](https://user-images.githubusercontent.com/87281783/235734225-344a6e7b-d3cb-45e5-9dff-4a6817a57b07.png)
Firefox Version: `Mozilla Firefox 112.0.2` (The latest version of firefox for arch linux)

It's still working for me. It could a error that affect some versions of firefox. Or it could be existing profile bug. Idk, however i'm sure about the problem is not from we.

> But we need to add new tweaks into `user.js` to hide bookmarks

**Comment by AlbertShown** [comment]
Yes, It seems to be a [Firefox issue](https://askubuntu.com/questions/1462267/when-launching-firefox-get-error-not-loading-module-atk-bridge), not WebUI.

You are right. The `user.js` and `userChrome.css` still need to be tweaked to make the Firefox window looks better.

**Comment by hassandraga** [comment]
Same issue on macOS.

![macOs_WebUI_Issue](https://user-images.githubusercontent.com/16948659/236607328-34dc04ba-523d-4783-bd04-6f06fa2dfe1a.png)

**Comment by hassandraga** [comment]
I'm not planning to fix this issue in v2.4.1. I hope someone have free time to deal with `user.js` and `userChrome.css` to make Firefox UI look better.

**Comment by malisipi** [comment]
Webview is good idea however there are a lot of Webview library in Linux. GTK' webview, QT' webview and more. It would be difficult.

We can download Chromium browser in background by library and run application with the browser.

However i think we should support more browsers to have more options. So, I think might be reconsidered Gnome Web https://github.com/webui-dev/webui/pull/225. (Epiphany is a web browser that developed by Gnome Team.)

**Comment by wwderw** [comment]
> Webview is good idea however there are a lot of Webview library in Linux. GTK' webview, QT' webview and more. It would be difficult.

Statistically speaking the GTK one is the one that I have found most to use, when it's just regular users downloading that dependency.  Qt webview is usually those that are using the Qt toolkit for the application as well.

Personally, I like the direction of webview more, from my experience, webview apps tend to have half the memory usage just on a hello world compared to the browser/electron route, so already starting off with less memory consumption.  Make it more appealing for even the more trivial apps as well.

The biggest downside (and this is more for the app devs) is that given that they are different (which would happen if different users used different browsers as well)  on each platform, is that the UI isn't consistent.  Takes more time for the dev to fix, but that's doable.

**Comment by wwderw** [comment]
> When WebUI use the already running browser, I guess it use even similar/less memory than an WebView app, **+** a real web browser offer full web technology for complex projects. However, most apps needs just an HTML renderer and a JS engine, so WebView is already enough in most cases.

In my experience, the resource usage of the min example using the browser is 550MB+ and a firm 280MB for webview.  Now, I don't know if that is a Linux/Firefox issue.  Maybe a Chromium based on may do better, I don't know about that.  But there is a stark difference between the two.

I do like the fact that a webview looks more polished as a desktop app compared to the browser (again though, this may be a Firefox thing as well), and just don't have all the extra fluff that a desktop app doesn't need (dev tools, extensions, bookmarks etc).

Now, I do agree about what is available when using a full browser compared to a webview (and webview on Linux, versus Windows, versus Mac seem to have differences among themselves).

> I use WebUI in some professional projects, we use `webui_show_browser` with `ChromiumBased`, all UI are similar, except the macOS Dock icon issue, which is not a big deal.

Yes, if I knew everyone that was using my app was using one specific browser, yes I would expect that UI to be similar (unless some fork made some changes, but I doubt it with the rendering engine), but I would be talking about the UI differences between firefox versus chromium or if someone was using a webkit based browser etc.

**Comment by malisipi** [comment]
> except the macOS Dock icon issue, which is not a big deal.

I have a idea to fix the Dock icon. Can we symlink the web browser and run it in a application package to use custom icon? I can not test the idea since I am not have a computer that has MacOS.

Maybe I work the issue about Firefox CSS issue in free time. I have a few ideas about it.
